sklearn/linear_model/ridge.py

```diff
@@ -203,6 +195,23 @@ def _deprecate_dense_cholesky(solver):
     return solver
 
 
+def _rescale_data(X, y, sample_weight):
+    """Rescale data so as to support sample_weight"""
+    n_samples = X.shape[0]
+    sample_weight = sample_weight * np.ones(n_samples)
+    sample_weight = np.sqrt(sample_weight)
+    if sparse.issparse(X):
+        sw_matrix = sparse.dia_matrix((sample_weight, 0),
+                                      shape=(n_samples, n_samples))
+        X = safe_sparse_dot(sw_matrix, X)
+    else:
```

Comment:
I think the code in the `if` clause would already quite gracefully handle the `else` clause. I.e. using a sparse diagonal matrix for the sample weights should solve the problem for all cases. (Maybe I am missing an exception?)